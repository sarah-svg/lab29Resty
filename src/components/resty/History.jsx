import React from 'react';
import PropTypes from 'prop-types';

const History = ({ url, method, id }) => {
  return (
    <section id={id}>
      <p id={id}>{url}</p>
      <p id={id}>{method}</p>
    </section>
  );
};

History.propTypes = {
  url: PropTypes.string.isRequired,
  method: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired

};

export default History;

