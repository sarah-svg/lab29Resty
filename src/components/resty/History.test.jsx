import React from 'react';
import { render, cleanup } from '@testing-library/react';
import History from './History';

describe('displays history of queries', () => {
  afterEach(() => cleanup());
  it('renders data from query history', () => {
    const { asFragment } = render(<History url="" method="" id="" 
      onSubmit="" onChange=""  />);
    expect(asFragment()).toMatchSnapshot();
  });
});
