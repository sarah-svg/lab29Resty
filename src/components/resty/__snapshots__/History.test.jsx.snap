// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays history of queries renders data from query history 1`] = `
<DocumentFragment>
  <section
    id=""
  >
    <p
      id=""
    />
    <p
      id=""
    />
  </section>
</DocumentFragment>
`;
